packages/lib/gpc/src/gpcUtil.ts

```diff
@@ -405,71 +524,88 @@ export function GPCRequirements(
  * @param proofConfig the proof configuration
  * @returns a record mapping a list name to an array of identifiers representing entries
  * and tuples that lie in that list
- * @throws TypeError if `liesInLists` is empty
+ * @throws TypeError if `isMemberOf` is empty
  */
 export function listConfigFromProofConfig(
   proofConfig: GPCProofConfig
 ): GPCProofMembershipListConfig {
   // Find all [listName, entryID] pairs in proofConfig.pods
-  const entryLists: [PODName, PODEntryIdentifier][] = Object.keys(
-    proofConfig.pods
-  )
-    .map((podName) => {
-      const pod = proofConfig.pods[podName];
-      const listEntryPairs = Object.keys(pod.entries)
-        .map((entryName): [PODName, PODEntryIdentifier][] => {
-          const lists = pod.entries[entryName].liesInLists;
-          if (lists !== undefined && lists.length === 0) {
+  const entryLists: [PODName, PODEntryIdentifier][] = [];
+
+  for (const podName of Object.keys(proofConfig.pods)) {
+    const pod = proofConfig.pods[podName];
+
+    for (const entryName of Object.keys(pod.entries)) {
+      const lists = pod.entries[entryName].isMemberOf;
+
+      if (lists !== undefined) {
+        if (Array.isArray(lists)) {
+          if (lists.length === 0) {
             throw new TypeError(
               `The list of lists of valid values for ${podName}.${entryName} is empty.`
             );
           }
-          return lists === undefined
-            ? []
-            : lists.map((listName): [PODName, PODEntryIdentifier] => [
-                listName,
-                `${podName}.${entryName}`
-              ]);
-        })
-        .flat();
-      return listEntryPairs;
-    })
-    .flat();
+          for (const listName of lists ?? []) {
+            entryLists.push([listName, `${podName}.${entryName}`]);
+          }
+        } else {
+          entryLists.push([lists, `${podName}.${entryName}`]);
+        }
+      }
+    }
+  }
 
   // Find all [listName, tupleID] pairs in proofConfig.tuples
-  const tupleLists: [PODName, TupleIdentifier][] = Object.keys(
-    proofConfig.tuples ?? {}
-  )
-    .map((tupleName): [PODName, TupleIdentifier][] => {
-      const lists = (proofConfig.tuples ?? {})[tupleName].liesInLists;
-      if (lists !== undefined && lists.length === 0) {
-        throw new TypeError(
-          `The list of lists of valid values for tuple.${JSON.stringify(
-            tupleName
-          )} is empty.`
-        );
+  const tupleLists: [PODName, PODEntryIdentifier][] = [];
+
+  for (const tupleName of Object.keys(proofConfig.tuples ?? {})) {
+    const lists = (proofConfig.tuples ?? {})[tupleName].isMemberOf;
```

Comment:
Neat trick.  I think the `??` on this line is provably unnecessary (if you reach this line it means there's a non-empty object), but probably the type-checker can't figure that out which makes it necessary.